client/src/stores/collectionAttributesStore.ts

```diff
@@ -0,0 +1,43 @@
+import { defineStore } from "pinia";
+import Vue, { computed, ref } from "vue";
+
+import { DatasetCollectionAttributes } from "./services";
+import * as Service from "./services/datasetCollection.service";
+
+export const useCollectionAttributesStore = defineStore("collectionAttributesStore", () => {
+    const storedAttributes = ref<{ [key: string]: DatasetCollectionAttributes }>({});
+    const loadingAttributes = ref<{ [key: string]: boolean }>({});
+
+    const getAttributes = computed(() => {
+        return (hdcaId: string) => {
+            if (!storedAttributes.value[hdcaId]) {
+                Vue.set(storedAttributes.value, hdcaId, {});
+                fetchAttributes({ hdcaId });
+            }
+            return storedAttributes.value[hdcaId];
+        };
+    });
+
+    const isLoadingAttributes = computed(() => {
+        return (hdcaId: string) => {
+            return loadingAttributes.value[hdcaId] ?? false;
+        };
+    });
+
+    async function fetchAttributes(params: { hdcaId: string }) {
+        Vue.set(loadingAttributes.value, params.hdcaId, true);
+        try {
+            const attributes = await Service.fetchCollectionAttributes(params);
```

Comment:
Only importing the required functions from a module can improve tree-shaking.

```suggestion
            const attributes = await fetchCollectionAttributes(params);
```

Reply:
Good point! Thanks!